polus-imagej-plugins/polus-imagej-image-integral-plugin/src/main.py

```diff
@@ -0,0 +1,164 @@
+import jpype, imagej, scyjava
+import typing
+
+# Initialize the logger
+logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("main")
+logger.setLevel(logging.INFO)
+
+def main(_opName: str,
+         _in1: Path,
+         _out: Path,
+         ) -> None:
+             
+    """ Initialize ImageJ """ 
+    # Bioformats throws a debug message, disable the loci debugger to mute it
+    def disable_loci_logs():
+        DebugTools = scyjava.jimport("loci.common.DebugTools")
+        DebugTools.setRootLevel("WARN")
+    scyjava.when_jvm_starts(disable_loci_logs)
+    
+    # This is the version of ImageJ pre-downloaded into the docker container
+    logger.info('Starting ImageJ...')
+    ij = imagej.init("sc.fiji:fiji:2.1.1+net.imagej:imagej-legacy:0.37.4",
+                     headless=True)
+    ij_converter.ij = ij
+    logger.info('Loaded ImageJ version: {}'.format(ij.getVersion()))
+    
+    """ Validate and organize the inputs """
+    args = []
+    arg_types = []
+    arg_len = 0
+
+    
+    # Validate opName
+    opName_values = [
+        "DefaultIntegralImg",
+        "SquareIntegralImg",
+        "WrappedIntegralImg",
+    ]
+    assert _opName in opName_values, 'opName must be one of {}'.format(opName_values)
+    
+    # Validate in1
+    in1_types = { 
+        "DefaultIntegralImg": "RandomAccessibleInterval",
+        "SquareIntegralImg": "RandomAccessibleInterval",
+        "WrappedIntegralImg": "RandomAccessibleInterval",
+    }
+    
+    # Check that all inputs are specified 
+    if _in1 == None and _opName in list(in1_types.keys()):
+        raise ValueError('{} must be defined to run {}.'.format('in1',_opName))
+    elif _in1 != None:
+        in1_type = in1_types[_opName]
+    
+        # switch to images folder if present
+        if _in1.joinpath('images').is_dir():
+            _in1 = _in1.joinpath('images').absolute()
+
+        args.append([f for f in _in1.iterdir() if f.is_file()])
+        arg_len = len(args[-1])
+    else:
+        arg_types.append(None)
+        args.append([None])
+    
+    for i in range(len(args)):
+        if len(args[i]) == 1:
+            args[i] = args[i] * arg_len
+            
+    """ Set up the output """
+    out_types = { 
+        "DefaultIntegralImg": "RandomAccessibleInterval",
+        "SquareIntegralImg": "RandomAccessibleInterval",
+        "WrappedIntegralImg": "RandomAccessibleInterval",
+    }
+
+    """ Run the plugin """
+    try:
+        for ind, (in1_path,) in enumerate(zip(*args)):
+            if in1_path != None:
+
+                # Load the first plane of image in in1 collection
+                logger.info('Processing image: {}'.format(in1_path))
+                in1_br = BioReader(in1_path)
+                
+                # Convert to appropriate numpy array
+                in1 = ij_converter.to_java(np.squeeze(in1_br[:,:,0:1,0,0]),in1_type)
```

Comment:
It doesn't look like the third argument of `to_java` is ever used. Why is it there? I've left comments on that function. More clarity should be added.